Subject: ps3: Distribute interrupts across CPU threads

From: Geert Uytterhoeven <Geert.Uytterhoeven@sonycom.com>

Distribute interrupts across CPU threads if no explicit binding is requested,
instead of binding them always to the first CPU thread.

Signed-off-by: Geert Uytterhoeven <Geert.Uytterhoeven@sonycom.com>
---
Question: is smp_processor_id() a good random distributor?

 arch/powerpc/platforms/ps3/interrupt.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/powerpc/platforms/ps3/interrupt.c
+++ b/arch/powerpc/platforms/ps3/interrupt.c
@@ -176,7 +176,7 @@ static int ps3_virq_setup(enum ps3_cpu_b
 	/* This defines the default interrupt distribution policy. */
 
 	if (cpu == PS3_BINDING_CPU_ANY)
-		cpu = 0;
+		cpu = smp_processor_id();
 
 	pd = &per_cpu(ps3_private, cpu);
 
